libtenzir/builtins/formats/bitz.cpp

```diff
@@ -0,0 +1,139 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/data.hpp>
+#include <tenzir/make_byte_reader.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/series_builder.hpp>
+#include <tenzir/table_slice.hpp>
+#include <tenzir/to_lines.hpp>
+
+namespace tenzir::plugins::bitz {
```

Comment:
The format is simple enough that creating a small ASCII diagram would help the reader. Idea:

```
 0                   1                   2                   3   
 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                           Version                             |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                            Size                               |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                                                               |
/                        Binary Buffer                          /
/                                                               /
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Reply:
I chose to add this instead of adding an ASCII diagram, as it is much more precise:

```cpp
// The BITZ format is a size-prefixed dump of Tenzir's wire format as laid out
// in the tenzir.fbs.FlatTableSlice FlatBuffers table. The size prefix occupies
// 64 bit and is stored in network byte order.
```

Reply:
🤷